Raven.Client.Lightweight/Shard/BaseShardedDocumentSession.cs

```diff
@@ -304,5 +305,21 @@ public IdToLoad(string id, IList<TDatabaseCommands> shards)
             public readonly string Id;
             public readonly IList<TDatabaseCommands> Shards;
         }
+
+        internal class ShardsOperation : Operation
+        {
+            internal Operation[] ShardsOperations { get; set; }
+            internal ShardsOperation(long id, RavenJToken state) : base(id, state)
+            {
+
+            }
+            public override RavenJToken WaitForCompletion()
```

Comment:
Adi, you need to override WaitForCompletionAsync, not the sync version, look here sync executes async: https://github.com/ayende/ravendb/blob/v3.5/Raven.Client.Lightweight/Connection/Operation.cs#L69